### PR TITLE
Adds on_gem_received hook [#258]

### PR DESCRIPTION
**Pull request by krisleech**
This a quick sketch of what I was thinking, no tests yet. Do you think this is reasonable so far?

### CODE REVIEW

**Comment by krisleech** [comment]
I'd use it something like:

config.ru

```ruby
class NotifyChat
  def call(gem)
    room.notify("A new gem was released: #{gem.spec.name} #{gem.spec.version}"
  end

  private

  def room
    # ...
  end
end

Geminabox.on_gem_received = NotifyChat.new
```

**Comment by sonots** [comment]
looks good to me, could you write some tests?

File: lib/geminabox/server.rb [hunk 1]
```diff
@@ -195,6 +195,10 @@ def handle_incoming_gem(gem)
         GemStore.create(gem, params[:overwrite])
       rescue GemStoreError => error
         error_response error.code, error.reason
+      else
+        if Geminabox.on_gem_received
+          Geminabox.on_gem_received.call(gem)
+        end
```
Comment (sonots) [inline lib/geminabox/server.rb]: We should think about some specifications:

1. return error_response if the callback errors
    * we can tell as the hook writer is responsible to the error handling, but I think it is safer to take care in geminabox codes
2. If the callback takes long time, the web server is occupied by the process. I think we have two options:
    * introduce some asynchronous library, and execute the callback asynchronously
    * Or, we can tell as the hook writer is responsible to the asynchronous execution.

Comment (sonots) [reply lib/geminabox/server.rb]: Sounds reasonable to me. Could you write about that on README and fix codes (rescuing error silently)?

**Comment by sonots** [comment]
I've written about warning on README and fixed codes to rescue error silently.

**Comment by krisleech** [comment]
Excellent, time to upgrade and try it out :smile:

**Pull request merged**